src/sentry/static/sentry/app/utils/advancedAnalytics.tsx

```diff
@@ -0,0 +1,107 @@
+import * as qs from 'query-string';
+
+import {Organization} from 'app/types';
+import {trackAnalyticsEvent} from 'app/utils/analytics';
+import {uniqueId} from 'app/utils/guid';
+import {
+  eventNameMap as integrationEventMap,
+  EventParameters as IntegrationEventParameters,
+} from 'app/utils/integrationEvents';
+
+const ANALYTICS_SESSION = 'ANALYTICS_SESSION';
+
+export const startAnalyticsSession = () => {
+  const sessionId = uniqueId();
+  window.sessionStorage.setItem(ANALYTICS_SESSION, sessionId);
+  return sessionId;
+};
+
+export const clearAnalyticsSession = () => {
+  window.sessionStorage.removeItem(ANALYTICS_SESSION);
+};
+
+export const getAnalyticsSessionId = () =>
+  window.sessionStorage.getItem(ANALYTICS_SESSION);
+
+const hasAnalyticsDebug = () => window.localStorage.getItem('DEBUG_ANALYTICS') === '1';
+
+//TODO: add more types/sources
+export type EventParameters = IntegrationEventParameters;
+
+export const allEventMap = {...integrationEventMap};
+
+type AnalyticsKey = keyof EventParameters;
+
+/**
+ * Tracks an event for analytics
+ * Must be tied to an organization
+ * Uses the current session ID or generates a new one if startSession == true
+ * An analytics session corresponds to a single action funnel such as installation
+ * Tracking by session allows us to track individual funnel attempts for a single user
```

Comment:
Minor nit but this documentation could use a bit of formatting, right now there's no periods anywhere or blank lines between sections, so it just reads like a big run on sentence